app/assets/javascripts/ui_components/modal.coffee

```diff
@@ -0,0 +1,13 @@
+$ ->
+  _.each $('.ui-components-modal'), (modal) ->
+    $modal = $ modal
+    trigger = $($modal.attr('data-trigger'))
+    return if trigger.length == 0
+    $(trigger).click (e) ->
+      url = $modal.attr('data-url')
+      if url && url.length > 0
+        $modal.find('.modal-body').load url, ->
+          $modal.one('shown.bs.modal', (e) -> $modal.trigger('uic:domchange'))
```

Comment:
The content is rendered before the modal is shown, but the modal takes some time after `.modal('show')` until everything is in a state where you can actually use it. That's the whole point of the `shown.bs.modal` event. We need this for the Markdown rendering and the date selector (so far) which is not yet implemented.